ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LedgerStateAccess.scala

```diff
@@ -66,10 +67,9 @@ trait LedgerStateOperations[LogResult] {
 /**
   * Convenience class for implementing read and write operations on a backing store that supports batched operations.
   */
-abstract class BatchingLedgerStateOperations[LogResult](implicit executionContext: ExecutionContext)
-    extends LedgerStateOperations[LogResult] {
+abstract class BatchingLedgerStateOperations[LogResult] extends LedgerStateOperations[LogResult] {
   override final def readState(key: Key): Future[Option[Value]] =
-    readState(Seq(key)).map(_.head)
+    readState(Seq(key)).map(_.head)(DirectExecutionContext)
```

Comment:
I'm not sure why you're removing the possibility to configure the execution context for the ledger state externally and even less why the execution context has to be a direct one. Can you elaborate?

Reply:
This is not for actually reading the state; that gets the execution context from the underlying implementation. (Remind me to rant about why futures are broken in Scala one day.)

This is just specifying the context for the `.map(_.head)` operation. The direct execution context basically says "use whatever context you used for the previous operation" (similar to [`parasitic` from Scala 2.13](http://www.scala-lang.org/api/2.13.3/scala/concurrent/ExecutionContext$$parasitic$.html), but less nuanced; however, the docs there still apply).

This means that we don't need to provide an execution context to the `BatchingLedgerStateOperations` class, because it'll re-use whatever the underlying implementation uses.

Reply:
Right, I see now it is applied to `map`.

Reply:
Note that there is some non-determinism in the `DirectExecutionContext` (see this discussion on [StackOverflow](https://stackoverflow.com/questions/61792512/non-determinism-of-synchronous-execution-contexts-a-k-a-parasitic)).

Reply:
Absolutely, but I think in this case it's all good.